src/zenml/integrations/mlflow/mlflow_utils.py

```diff
@@ -100,17 +100,18 @@ def enable_mlflow_run(run: Callable[..., Any]) -> Callable[..., Any]:
     """
 
     def inner_decorator(
-        self: BasePipeline, run_name: Optional[str] = None
+        self: BasePipeline, *, run_name: Optional[str] = None, **kwargs: Any
     ) -> Any:
         """Inner decorator used to extend the run method of a pipeline.
         This ensures each pipeline run is run within a different mlflow context.
 
         Args:
             self: self of the original pipeline class
             run_name: Optional name for the run.
+            **kwargs: Additional kwargs passed to `pipeline.run()`
         """
         with mlflow.start_run(run_name=run_name):
-            run(self, run_name)
+            return run(self, run_name=run_name, **kwargs)
```

Comment:
I am not too familiar with decorators - especially layerd like this - how come it worked before without returning the function? And why is it better here to return it?

Reply:
Essentially this makes sure the new `pipeline.run` method (the `inner_decorator`) returns the return value of the overwritten default `pipeline.run` method. In the local/kubeflow case this didn't matter as the method didn't return anything, but for airflow it returns an airflow object which gets forwarded here.

Reply:
Thanks for the explanation, makes sense